pkg/cloud/aws/converters/instance.go

```diff
@@ -36,14 +38,12 @@ func SDKToInstance(v *ec2.Instance) *v1alpha1.Instance {
 		EBSOptimized: v.EbsOptimized,
 	}
 
+	i.IAMProfile = strings.Split(aws.StringValue(v.IamInstanceProfile.Arn), "instance-profile/")[1]
+
 	for _, sg := range v.SecurityGroups {
 		i.SecurityGroupIDs = append(i.SecurityGroupIDs, *sg.GroupId)
 	}
 
-	// TODO: Handle returned IAM instance profile, since we are currently
-	// using a string representing the name, but the InstanceProfile returned
-	// from the sdk only returns ARN and ID.
```

Comment:
We should be able to generate the IAM ARN given a name, the account-id can be queried at runtime

Reply:
Are you suggesting that a better flow would be to look up the Account ID, generate an ARN given the name passed into the instance, and then compare that to the full ARN returned by the EC2 query?

Reply:
We could do that instead and it'd save us the trouble of splitting the string returned from the API and have a more consistent behavior, a TODO comment should do it, we don't have to implement this here.

Reply:
Understood. Thanks for clarifying! Comment added.

Reply:
> We should be able to generate the IAM ARN given a name, the account-id can be queried at runtime

This isn't strictly true. IAM objects can be referenced by the "Friendly Name" as they are unique per-object type-per-account, but as an ARN, an IAM object can be stored in a path. This is used for scoping down permissions. For example:

An instance profile stored at:
`arn:aws:iam::123456789012:instance-profile/application_abc/example-profile` can be referenced simply as `example-profile` or the ARN, but `arn:aws:iam::123456789012:instance-profile/example-profile` would result in a not found.

You would need to resolve the ARN of the friendly name using IAM::GetInstanceProfile

Reply:
To clarify, would `foo/bar` be valid as a friendly name? I'm curious as to why having paths would make generating an ARN from friendly name more difficult.

Reply:
So would it be best to update the TODO to reflect that the best end state would be to look up the instance profile by name and compare the ARNs?

Reply:
Pretty much. I guess the analogues AWS are using is between a fully qualified DN including the Organisational Unit path and a plain Username in LDAP.

Reply:
Makes sense to me. Thanks for the clarification!